AsyncDisplayKit/Details/ASCollectionDataController.mm

```diff
@@ -28,22 +28,30 @@ @implementation ASCollectionDataController {
   NSMutableDictionary *_pendingIndexPaths;
 }
 
-- (void)prepareForReloadData
+- (instancetype)initWithAsyncDataFetching:(BOOL)asyncDataFetchingEnabled
 {
-  _pendingNodes = [NSMutableDictionary dictionary];
-  _pendingIndexPaths = [NSMutableDictionary dictionary];
+  self = [super initWithAsyncDataFetching:asyncDataFetchingEnabled];
+  if (self != nil) {
+    _pendingNodes = [NSMutableDictionary dictionary];
+    _pendingIndexPaths = [NSMutableDictionary dictionary];
+  }
+  return self;
+}
 
-  [[self supplementaryKinds] enumerateObjectsUsingBlock:^(NSString *kind, NSUInteger idx, BOOL *stop) {
+- (void)prepareForReloadData
```

Comment:
The pending dictionaries exist to keep the pending state between the `prepare` and `will` methods. I've set them up to be created at initialization such that they'll remain throughout the lifetime of the collection data controller. Their contents are cleared out at the end of the `will` methods.